src/ui/components/SecondaryToolbox/ReduxDevTools.tsx

```diff
@@ -75,35 +75,50 @@ export const ReduxDevToolsPanel = () => {
     const rootComponent = rootRef.current;
     const store = rootComponent.store!;
 
-    const matchingAnnotationsByTimeRange = reduxAnnotations.filter(annotation => {
-      return currentTimestamp != null && annotation.time < currentTimestamp;
+    const allInstanceIds: string[] = ["default"];
+
+    reduxAnnotations.forEach(annotation => {
+      annotation.action.request.instanceId;
+      // @ts-ignore ignore "this comparison never matches" error
+      if (annotation.action?.request?.type === "INIT_INSTANCE") {
+        // @ts-ignore-error TS is _really_ getting confused here
+        allInstanceIds.push(annotation.action.request.instanceId);
+      }
     });
 
-    if (!matchingAnnotationsByTimeRange.length) {
-      return;
-    }
-
     batch(() => {
+      // We may receive the Redux annotations in several groups over time.
+      // Rather than try to figure out which ones we have and haven't loaded into
+      // the RDT Redux store, we'll just try to reset its internal instance state entirely,
+      // and always reload all of the received annotation actions.
+
       // TODO THIS IS A TERRIBLE IDEA COME UP WITH SOMETHING BETTER AND YET THIS WORKS
-      // Seemingly reset the DevTools internal knowledge of actions,
-      // so we can re-send in all actions up to this point in the recording
+      // Seemingly reset the DevTools internal knowledge of actions.
+      // @ts-ignore this is legal to dispatch
       store.dispatch({
-        type: "devTools/UPDATE_STATE",
-        request: {
-          type: "LIFTED",
-          id: "default",
-        },
+        type: "socket/DISCONNECTED",
+      });
+
+      // For each connection instance that we know will exist from annotations,
+      // re-create an empty state value in the instances data
+      allInstanceIds.forEach(instanceId => {
+        store.dispatch({
+          type: "devTools/UPDATE_STATE",
+          request: {
+            type: "LIFTED",
+            id: instanceId,
+          },
+        });
       });
 
-      // TODO Is this how we want to actually behave here?
-      // Update the store with the actions up to this point in time
-      matchingAnnotationsByTimeRange.forEach(annotation => {
+      // Originally I had this showing only actions up to the current pause point,
+      // but that led to a situation where there could be partly-initialized state and
+      // the RDT UI would end up crashing. Now, always load everything we've received.
+      reduxAnnotations.forEach(annotation => {
```

Comment:
Won't this double-dispatch the "INIT_INSTANCE" actions? Since we also dispatch them above? Or is that okay?

Reply:
I'm not actually _dispatching_ the `INIT_INSTANCE` actions above - just trying to identify them so I can ensure that we have an empty but valid initial set of state data for each one.

But, beyond that, the intent here is to fully reset the internals of the `instances` reducer every time this logic runs, so that we reload the actions data from scratch.

Reply:
> I'm not actually dispatching the INIT_INSTANCE actions above

Aren't you though? I'm referring to this:
```js
        store.dispatch({
          type: "devTools/UPDATE_STATE",
          request: {
            type: "LIFTED",
            id: instanceId,
          },
        });
```

Reply:
Nevermind my previous comment. I see what you're saying now about the dispatch being different values, though I still don't understand _why_ it's needed 😁